src/main/java/org/killbill/billing/plugin/adyen/api/mapping/UserDataMappingService.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2014-2015 Groupon, Inc
+ * Copyright 2014-2015 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.killbill.billing.plugin.adyen.api.mapping;
+
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+import org.killbill.billing.account.api.Account;
+import org.killbill.billing.payment.api.PluginProperty;
+import org.killbill.billing.plugin.adyen.client.model.UserData;
+import org.killbill.billing.plugin.api.PluginProperties;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Optional;
+
+public class UserDataMappingService {
```

Comment:
[Note for later] This seems like a good candidate to extract to https://github.com/killbill/killbill-plugin-framework-java, as other plugins need to do the same.